cli/src/pcluster/schemas/cluster_schema.py

```diff
@@ -129,135 +415,205 @@ def make_config(self, data, **kwargs):
-            mount_dir = input_data["MountDir"]
-            input_data["Settings"].update({"MountDir": mount_dir})
-            input_data[f"{storage_type}Settings"] = input_data.pop("Settings")
-
-            del input_data["MountDir"]
-            del input_data["StorageType"]
-        else:
-            # raise error
-            raise ValidationError("Wrong storage type")
+    @post_load
+    def make_config(self, data, **kwargs):
+        """Generate config object."""
+        return LogsConfig(**data)
 
-        return input_data
+
+class DashboardsSchema(BaseSchema):
+    """Represent the schema of the SharedStorage with type = EFS."""
+
+    cloud_watch = fields.Nested(CloudWatchDashboardsSchema)
 
     @post_load
     def make_config(self, data, **kwargs):
-        """Return the right SharedStorage type object, according to the given type."""
-        if "ebs_config" in data:
-            return data["ebs_config"]
-        elif "efs_config" in data:
-            return data["efs_config"]
-        elif "fsx_config" in data:
-            return data["fsx_config"]
+        """Generate config object."""
+        return DashboardsConfig(**data)
+
+
+class MonitoringSchema(BaseSchema):
+    """Represent the schema of the SharedStorage with type = EFS."""
+
+    detailed_monitoring = fields.Bool()
+    logs = fields.Nested(LogsSchema)
+    dashboards = fields.Nested(DashboardsSchema)
+
+    @post_load
+    def make_config(self, data, **kwargs):
+        """Generate config object."""
+        return MonitoringConfig(**data)
+
+
+# ---------------------- Others ---------------------- #
+class TagSchema(BaseSchema):
+    """Represent the schema of tag."""
+
+    key = fields.Str(data_key="Key")
+    value = fields.Str(data_key="Value")
+
+    @post_load
+    def make_config(self, data, **kwargs):
+        """Generate config object."""
+        return TagConfig(**data)
+
+
+class RolesSchema(BaseSchema):
+    """Represent the schema of roles."""
+
+    head_node = fields.Str()
+    compute_node = fields.Str()
+    custom_lambda_resources = fields.Str()
+
+    @post_load
+    def make_config(self, data, **kwargs):
+        """Generate config object."""
+        return RolesConfig(**data)
+
+
+class S3AccessSchema(BaseSchema):
+    """Represent the schema of S3 access."""
+
+    bucket_name = fields.Str(required=True)
+    type = fields.Str()
+
+    @post_load
+    def make_config(self, data, **kwargs):
+        """Generate config object."""
+        return S3AccessConfig(**data)
 
 
+class AdditionalIamPolicySchema(BaseSchema):
+    """Represent the schema of Additional IAM policy."""
+
+    policy = fields.Str(required=True)
+    scope = fields.Str()
+
+    @post_load
+    def make_config(self, data, **kwargs):
+        """Generate config object."""
+        return AdditionalIamPolicyConfig(**data)
+
+
+class IamSchema(BaseSchema):
+    """Represent the schema of IAM."""
+
+    roles = fields.Nested(RolesSchema, data_key="Roles")
```

Comment:
minor: you can avoid the data_key

Reply:
Done, also cleaned up similar issues elsewhere